Add GCP information to existing Astro documentation (#661)
astronomer/docs: astro/create-cluster.md

description: Create a new Astro Cluster for your Organization.
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Overview

The Astro install typically starts with 1 Cluster per Organization. Depending on your use case, however, your team can choose to configure multiple Astro Clusters. This could enable a few benefits, including:

- Clusters in different regions

> - Different Clusters for Development and Production

Reviewer: ```suggestion
- Different Clusters for development and production environments
```

> Within a single Workspace, you can host Deployments across multiple Clusters. For example, you can have Production Deployments running in your Production Cluster and Development Deployments running in your Development Cluster.

Reviewer: ```suggestion
Within a single Workspace, you can host Deployments across multiple Clusters. For example, you might have a production Deployment running in a production Cluster and a development Deployment running in a development Cluster. Both of those Deployments can be in the same Workspace.
```

> To complete this setup, you need to have:

Reviewer: ```suggestion
To create an Astro Cluster, you need to have:
```

> - An Astronomer installation.

Reviewer: ```suggestion
- An activated Data Plane
```

Toying with words here but I don't wanna add "installation" as another word for "Cluster", "Data Plane", "Organization"

> - Permissions to edit trust policies in the dedicated account for Astro on your cloud.

Reviewer 2: There's no real single word alternative to Trust Policies for GCP.  How about we just keep it really neutral "permissions to edit IAM in the dedicated account/project for Astro"

> - Permissions to edit trust policies in the dedicated account for Astro on your cloud.

Reviewer: ```suggestion
- Permissions to edit IAM for the dedicated Astro account on your cloud. If AWS, this is trust policies.
```

Author: I think it's okay to not mention AWS specifically here

<Tabs
    defaultValue="aws"
    values={[
        {label: 'AWS, value: 'aws'},
        {label: 'GCP', value: 'gcp'},
    ]}>
<TabItem value="aws">

#### Step 1: Submit a Request to Astronomer

> To create a new Cluster in your Organization, you must first reach out to your Astronomer representative. For each new Cluster that you want to provision, you'll need to provide our team with the following information:

Reviewer: ```suggestion
To create a new Astro Cluster on AWS for your Organization, you must first reach out to your Astronomer representative. For each new Cluster that you want to provision, provide our team with the following information:
```

> From there, your Astronomer representative will provide you with a unique `External ID` for each new Cluster. Make note of this value for the next step.

Reviewer: ^Can't make comment but let's remove `the default is 20` and copy sentence from GCP section on saying the default below

    aws iam set-security-token-service-preferences --global-endpoint-token-version v2Token
    ```

#### Step 3: Confirm with Astronomer

> Astronomer will notify you once your cluster has been created. Once your Cluster is available, you should be able to [configure a new Deployment](configure-deployment.md) within that cluster.

Reviewer: ```suggestion
Astronomer will notify you once your Cluster has been created. Once your Cluster is available, you should be able to [configure a new Deployment](configure-deployment.md) within that Cluster.
```

</TabItem>

<TabItem value="gcp">

#### Step 1: Submit a Request to Astronomer

> To create a new Cluster in your Organization, you must first reach out to your Astronomer representative. For each new Cluster that you want to provision, you'll need to provide our team with the following information:

Reviewer: ```suggestion
To create a new Astro Cluster on GCP for your Organization, you must first reach out to your Astronomer representative. For each new Cluster that you want to provision, provide our team with the following information:
```

- Your preferred Astro Cluster name.
- The GCP region that you want to host your Cluster in.
- Your preferred node instance type.
- Your preferred CloudSQL instance type.
- Your preferred maximum node count.

> If not specified, Astronomer will create a Cluster with two e2-medium-4 nodes, one Medium General Purpose CloudSQL instance (4vCPU, 16GB), and a maximum node count of 20 in `us-central1`.

Reviewer 2: Can you please remove the number two?  We actually create three nodes, but its complicated - we use n2 for the system pool and e2 for the worker pool.  So, if we just say "Astronomer will create a Cluster with e2-medium-4 nodes" that gives us some flexibility because this will rapidly change as we add worker queue support.

> If not specified, Astronomer will create a Cluster with two e2-medium-4 nodes, one Medium General Purpose CloudSQL instance (4vCPU, 16GB), and a maximum node count of 20 in `us-central1`.

Reviewer: ```suggestion
If not specified, Astronomer will create a Cluster with `e2-medium-4 nodes`, one Medium General Purpose CloudSQL instance (4vCPU, 16GB), and a maximum node count of 20 in `us-central1`.
```

For information on all supported regions and configurations, see [GCP Resource Reference](resource-reference-gcp.md).  

#### Step 2: Confirm with Astronomer

> Astronomer will notify you once your cluster has been created. Once your Cluster is available, you should be able to [configure a new Deployment](configure-deployment.md) within that cluster.

Reviewer: ```suggestion
Astronomer will notify you once your Cluster has been created. Once your Cluster is available, you should be able to [configure a new Deployment](configure-deployment.md) within that Cluster.
```
